src/esp/gfx/replay/Keyframe.h

```diff
@@ -39,18 +39,43 @@ struct RenderAssetInstanceState {
   }
 };
 
+/**
+ * @brief Registration of a rigged bone.
+ */
+// TODO: Serializer doesn't support collections in nested types. Therefore,
+// rigId is duplicated for each bone.
+struct BoneCreation {
+  int rigId = ID_UNDEFINED;
+  int boneId = ID_UNDEFINED;
+  std::string boneName;
+};
+
+/**
+ * @brief The dynamic state of a rigged object bone that is tracked by the
+ * replay system every frame.
+ */
+// TODO: Serializer doesn't support collections in nested types. Therefore,
```

Comment:
This isn't my understanding. I thought our serialization code could handle arbitrary nesting of all serialization-supported types including e.g. vectors of vectors or vectors of objects containing vectors. What nesting did you want to do and what error did you hit?

Reply:
I wanted to serialize `RigCreations` and `RigUpdates` structs rather than individual bones to simplify the code and avoid writing one `rigId` per bone entry.

Both of these would be structs, with collections of bones and rig-specific data if needed.

My understanding is that the json serialization templating breaks down when a top-level object has a collection of structs, which themselves have collections. This applies to vectors, optionals, and others collections.

I considered supporting that outside of the scope of this changeset. I would focus on making the feature work and revisit serialization altogether when we look into optimization.

@jturner65 can you chime in here?